src/triage/component/catwalk/storage.py

```diff
@@ -213,123 +243,179 @@ def matrix_with_sorted_columns(self, columns):
                     Columnset and desired columnset mismatch. Unique items: %s
                 ''', columnset ^ desired_columnset)
 
-    def save_yaml(self, df, project_path, name):
-        with smart_open.smart_open(os.path.join(project_path, name + ".yaml"), "wb") as f:
-            yaml.dump(df, f, encoding='utf-8')
+    def save_metadata(self, df, project_path, name):
+        path_parsed = urlparse(project_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+        if not scheme or scheme == 'file':  # Local file
+            with open(os.path.join(project_path, name + ".yaml"), "wb") as f:
+                yaml.dump(df, f, encoding='utf-8')
+        elif scheme == 's3':
+            s3 = s3fs.S3FileSystem()
+            with s3.open(os.path.join(project_path, name + ".yaml"), "wb") as f:
+                yaml.dump(df, f, encoding='utf-8')
+        else:
+            raise ValueError(f"""
+                  URL scheme not supported:
+                  {scheme} (from {os.path.join(project_path, name + '.yaml')})
+            """)
+
+    def load_metadata(self):
+        path_parsed = urlparse(self.metadata_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+        if not scheme or scheme == 'file':  # Local file
+            with open(self.metadata_path, "r", encoding='utf-8') as f:
+                metadata = yaml.load(f.read())
+        elif scheme == 's3':
+            s3 = s3fs.S3FileSystem()
+            with s3.open(self.metadata_path, 'rb', encoding='utf-8') as f:
+                metadata = yaml.load(f.read())
+        else:
+            raise ValueError(f"""
+                  URL scheme not supported:
+                  {scheme} (from {self.metadata_path})"
+            """)
 
-    def load_yaml(self, metadata_path):
-        with smart_open.smart_open(metadata_path, "rb") as f:
-            y = []
-            for line in f:
-                y.append(line.decode())
-        return yaml.load("".join(y).encode('utf-8'))
+        return metadata
 
     def __getstate__(self):
         # when we serialize (say, for multiprocessing),
         # we don't want the cached members to show up
         # as they can be enormous
-        self._matrix = None
+        self.matrix = None
         self._labels = None
-        self._metadata = None
-        self._head_of_matrix = None
+        self.metadata = None
+        self.head_of_matrix = None
```

Comment:
I'm not sure that we want to clear the object's cache, so much as omit its cached members from the pickle. It might not make a difference due to how it's used, of course. But it seems simple enough to say –

    nopickle = frozenset(['metadata', 'matrix', 'head_of_matrix'])
    return {key: value for (key, value) in self.__dict__.items() if key not in nopickle}

For that matter, you might be able to get really fancy (and DRY) –

    nopickle = {name for (name, item) in self.__class__.__dict__.items()
                if isinstance(item, cachedproperty)}
    …